pyfar/signal.py

```diff
@@ -379,57 +143,25 @@ def comment(self):
     @comment.setter
     def comment(self, value):
         """Set comment."""
-        self._comment = str(value)
+        self._comment = 'none' if value is None else str(value)
 
     def copy(self):
         """Return a deep copy of the Signal object."""
         return utils.copy(self)
 
-    def __add__(self, data):
-        return add((self, data), 'freq')
-
-    def __sub__(self, data):
-        return subtract((self, data), 'freq')
-
-    def __mul__(self, data):
-        return multiply((self, data), 'freq')
+    def __getitem__(self, key):
+        """Get signal channels at key.
+        """
+        if isinstance(key, (int, slice, tuple)):
+            try:
+                data = self._data[key]
+            except KeyError:
+                raise KeyError("Index is out of bounds")
+        else:
+            raise TypeError(
+                    "Index must be int, not {}".format(type(key).__name__))
 
-    def __truediv__(self, data):
-        return divide((self, data), 'freq')
-
-    def __pow__(self, data):
-        return power((self, data), 'freq')
-
-    def __repr__(self):
-        """String representation of signal class.
-        """
-        repr_string = (
-            f"{self.domain} domain {self.signal_type} Signal:\n"
-            f"{self.cshape} channels with {self.n_samples} samples @ "
-            f"{self._sampling_rate} Hz sampling rate and {self.fft_norm} "
-            "FFT normalization\n")
-
-        return repr_string
-
-    def __getitem__(self, key):
-        """Get signal channels at key.
-        """
-        if isinstance(key, (int, slice, tuple)):
-            try:
-                data = self._data[key]
-            except KeyError:
-                raise KeyError("Index is out of bounds")
-        else:
-            raise TypeError(
-                    "Index must be int, not {}".format(type(key).__name__))
-        items = Signal(
-            data,
-            sampling_rate=self.sampling_rate,
-            domain=self.domain,
-            fft_norm=self.fft_norm,
-            dtype=self.dtype)
-
-        return items
+        return self._return_item(data)
```

Comment:
`_return_item` is not a function of _Audio, so would it make sense to implement a default of it for _Audio and keep the redefinitions in the subclasses?

Reply:
Good point. I added empty methods for this and `assert_matching_meta_data` that raise `NotImplementedErrors` if called from the `Audio` base class. There are also tests for checking this :)